pkg/docker/containers-view.jsx

```diff
@@ -371,60 +417,98 @@ var ImageList = React.createClass({
         this.setState({ pulling: this.props.client.pulling });
     },
 
+    vulnerableInfoChanged: function (event, infos) {
+        this.setState({ vulnerableInfos: infos });
+    },
+
     componentDidMount: function () {
         $(this.props.client).on('image.containers', this.imagesChanged);
         $(this.props.client).on('pulling.containers', this.pullingChanged);
+
+        atomic.addEventListener('vulnerableInfoChanged', this.vulnerableInfoChanged);
     },
 
     componentWillUnmount: function () {
         $(this.props.client).off('image.containers', this.imagesChanged);
         $(this.props.client).off('pulling.containers', this.pullingChanged);
+
+        atomic.removeEventListener('vulnerableInfoChanged', this.vulnerableInfoChanged);
     },
 
-    render: function () {
-        var filtered = this.state.images.filter(function (image) {
-            return (image.RepoTags &&
-                    image.RepoTags[0].toLowerCase().indexOf(this.props.filterText.toLowerCase()) >= 0);
-        }.bind(this));
+    renderRow: function (image, withVul) {
+        var vulnerabilityColumn = '';
+
+        var vulnerableInfo = this.state.vulnerableInfos[image.Id.replace(/^sha256:/, '')];
+        if (vulnerableInfo) {
+            var count = vulnerableInfo.vulnerabilities.length;
```

Comment:
Variable hoisting is at play here. You can either use ```let``` or declare ```count``` at the function level.